Data/MySQL/src/SessionImpl.cpp

```diff
@@ -255,7 +256,16 @@ void SessionImpl::setTransactionIsolation(Poco::UInt32 ti)
 Poco::UInt32 SessionImpl::getTransactionIsolation() const
 {
 	std::string isolation;
-	getSetting("tx_isolation", isolation);
+	unsigned long version = Utility::serverVersion(_handle);
+	if (version >= 80000)
+	{
+		getSetting("transaction_isolation", isolation);
+		isolation = isolation.c_str();
```

Comment:
Trimming required because the buffer is the same size of the capacity after the getSetting() call. If this is not done then the comparisons below (e.g. MYSQL_READ_UNCOMMITTED == ...etc) will fail.